pkg/virt-api/rest/console.go

```diff
@@ -107,8 +107,12 @@ func (t *Console) Console(request *restful.Request, response *restful.Response)
 
 	dstAddr := string(pods.Items[0].Status.PodIP)
 
-	// FIXME, don't hardcode virt-handler port. virt-handler should register itself somehow
-	port := "8185"
+	port := string(pods.Items[0].Annotations["virtHandlerPort"])
```

Comment:
I think this all makes my brain hurt :D 

It's possible virt-handler may have to change network namespaces.  It's also possible we will want to introduce a networking component that lives in the host network namespace and just helps virt-handler coordinate some of the CNI stuff.   I'm really not sure yet.  I know you had some reservations about introducing another component dedicated to networking. That discussion is probably better left until another time.

@rmohr as far as this PR goes. I'm just pointing out that virt-handler doesn't have to (and probably shouldn't even have permission to) modify it's Pod definition to populate the port information.

If we want to pass the port using annotations, why not just explicitly set the annotation in the virt-handler manifest. We should know what the port is going to be up front when we define the virt-launcher daemon set.